### DISCUSSION TITLE
Is there a reason why the full DevTools Protocol isn't supported?

### DISCUSSION THREAD

**Discussion by postmodern** [rubycdp/ferrum#292] [General]
I noticed that many of the functions from the [Chrome DevTools Protocol (1.2)](https://chromedevtools.github.io/devtools-protocol/1-2/) (and 1.3) are missing from ferrum (ex: `Debugger`, `DOMDebugger`, most of the `Runtime` functions, `Browser.getVersion`, `Network.webSocket*`). Is this intentional or would you be open to Pull Requests to add them?

**Comment by route** [comment]
Hey, I don't mind adding any features as long as it fits ferrum's design. Please feel free!

**Reply by postmodern** [reply]
Would you also accept PRs that added YARD documentation tags? I noticed most of the classes and methods are undocumented. I prefer adding YARD tags to all new code that way it renders nicely on [rubydoc.info](https://rubydoc.info/gems/).

**Reply by route** [reply]
For sure! I just didn't do it from the beginning, and now it's just hard
